research-hub-web/src/app/app.component.ts

```diff
@@ -135,31 +230,33 @@ export class AppComponent implements OnInit, OnDestroy, AfterViewInit {
     this.appComponentService.setContentSidenavHasContent(hasContent);
   }
 
-  /**
-   * Sets the page title, and enables/disables the search bar and header.
-   * Also sets any custom CSS for the page.
-   * @param pageInfo currentPage information object.
-   */
-  setTitleSearchBarHeaderCustomCSS(pageInfo: any) {
-    if (pageInfo.title) {
-      this.titleService.setTitle('ResearchHub: ' + this.pageTitle);
-    }
-    this.headerService.setBatchParams(pageInfo.title, pageInfo.description, pageInfo.imageUrl, pageInfo.isHeaderVisible);
-    this.searchBarService.setVisibility(pageInfo.isSearchBarVisible);
-    this.appComponentService.setCustomCSSClassName(pageInfo.customCSSClassName);
-  }
+  async ngOnInit() {  
+    this.title = "Welcome to ResearchHub"
+    this.summary = "The ResearchHub connects you with people, resources, and services from across the University to enhance and accelerate your research."
+
+    // Methods dependent on URL
+    this.url = this.appComponentService.url.subscribe(url => {
+      if (url == 'home') {
+        this.titleService.setTitle('ResearchHub | Home');
+        this.showBanner = true;
+      }
+      else {
+        this.titleService.setTitle('ResearchHub | Home');
+        this.showBanner = false;
+      }
+      this.searchBarService.setVisibility(url == 'home' || url.substring(0, 6) == 'search' ? true : false);
```

Comment:
As mentioned below, you should delete all this.

We're already subscribing to route changes on line 273. So inside of that subscription, you can just put this search bar visibility line (although I recommend tidying it a little bit to something like this):

```ts
this.searchBarService.setVisibility(['home', 'search'].includes(routeName));
```